protocols/relay/src/behaviour/handler.rs

```diff
@@ -896,33 +943,34 @@ impl ConnectionHandler for Handler {
         >,
     ) {
         match event {
-            ConnectionEvent::FullyNegotiatedInbound(fully_negotiated_inbound) => {
-                self.on_fully_negotiated_inbound(fully_negotiated_inbound)
-            }
-            ConnectionEvent::FullyNegotiatedOutbound(fully_negotiated_outbound) => {
-                self.on_fully_negotiated_outbound(fully_negotiated_outbound)
+            ConnectionEvent::FullyNegotiatedInbound(FullyNegotiatedInbound {
+                protocol: stream,
+                ..
+            }) => {
+                self.inbound_requests_futs
+                    .push(self.process_inbound_request(stream));
```

Comment:
Instead of having one function that constructs a future and pushes into the `FuturesUnordered` here, I'd prefer to rename `process_inbound_request` to `on_fully_negotiated_inbound` and directly modify `self.inbound_requestes_futs` inside.